db/models.py

```diff
@@ -0,0 +1,51 @@
+from sqlalchemy import Column, String, DateTime, ForeignKey
+from sqlalchemy.dialects.postgresql import UUID, JSONB
+from datetime import datetime
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+from pydantic import BaseModel, ConfigDict
+
+Base = declarative_base()
+
+
+class UserOrm(Base):
+    __tablename__ = "users"
+
+    id = Column(String, primary_key=True, unique=True, nullable=False)
+    name = Column(String, nullable=False)
+    email = Column(String, unique=True, nullable=False)
+    created_at = Column(DateTime, nullable=False, default=datetime.now())
+    updated_at = Column(DateTime, nullable=False, default=datetime.now())
+    threads = relationship("ThreadOrm", back_populates="user")
+
+
+class ThreadOrm(Base):
+    __tablename__ = "threads"
+
+    id = Column(String, primary_key=True, unique=True, nullable=False)
+    user_id = Column(String, ForeignKey("users.id"), nullable=False)
+    created_at = Column(DateTime, nullable=False, default=datetime.now())
+    updated_at = Column(
+        DateTime, nullable=False, default=datetime.now(), onupdate=datetime.now()
+    )
+    content = Column(JSONB, nullable=False)
+    user = relationship("UserOrm", back_populates="threads")
+
+
+class ThreadModel(BaseModel):
+    model_config = ConfigDict(from_attributes=True)
+    id: str
+    user_id: str
+    created_at: datetime
+    updated_at: datetime
+    content: dict
+
+
+class UserModel(BaseModel):
+    model_config = ConfigDict(from_attributes=True)
+    id: str
+    name: str
+    email: str
+    created_at: datetime
+    updated_at: datetime
+    threads: list[ThreadModel] = []
```

Comment:
One question, why is this necessary? Isn't the `user_id` foreign key on the Threads table enough?

Reply:
The foreign key in the `UserOrm` will load the threads for a User, but the lines you've highlighted are for the pydantic UserModel - since the ORM only loads the relationship when accessed, we need to declare it as a field on the model. I'm following this logic: https://docs.pydantic.dev/latest/examples/orms/#sqlalchemy to integrate the pydantic models with the SQLAlchemy ORM models - there might be a neater way to do this